pkg/collector/process_test.go

```diff
@@ -120,19 +120,21 @@ func TestTCPCollectingProcess_ReceiveDataRecord(t *testing.T) {
 
 	// wait until collector is ready
 	waitForCollectorReady(t, cp)
+	var conn net.Conn
 	collectorAddr := cp.GetAddress()
 	go func() {
-		conn, err := net.Dial(collectorAddr.Network(), collectorAddr.String())
+		conn, err = net.Dial(collectorAddr.Network(), collectorAddr.String())
 		if err != nil {
 			t.Errorf("Cannot establish connection to %s", collectorAddr.String())
 		}
-		defer conn.Close()
 		conn.Write(validDataPacket)
 	}()
 	<-cp.GetMsgChan()
 	cp.Stop()
-	// Check if connection has closed properly or not by creating a new connection.
-	_, err = net.Dial(collectorAddr.Network(), collectorAddr.String())
+	// Check if connection has closed properly or not by writing to it
```

Comment:
We need to check the connection is closed properly and the collector server is properly stopped or not (server close), right? For the latter, we can still use the `Dial` as the check.